bin/hdfcoinc/pycbc_coinc_statmap

```diff
@@ -133,20 +212,301 @@ logging.info("calculating ifar/fap values")
 if fore_locs.sum() > 0:
     ifar = background_time / (fnlouder + 1)
     fap = 1 - numpy.exp(- coinc_time / ifar)
-    f['foreground/ifar'] = sec_to_year(ifar)
-    f['foreground/fap'] = fap
+    f['foreground_h/h_iter' + h_iter_string + '/ifar'] = sec_to_year(ifar)
+    f['foreground_h/h_iter' + h_iter_string + '/fap'] = fap
 
     ifar_exc = background_time_exc / (fnlouder_exc + 1)
     fap_exc = 1 - numpy.exp(- coinc_time_exc / ifar_exc)
     f['foreground/ifar_exc'] = sec_to_year(ifar_exc)
     f['foreground/fap_exc'] = fap_exc
 else:
-    f['foreground/ifar'] = numpy.array([])
-    f['foreground/fap'] = numpy.array([])
-    f['foreground/ifar_exc'] = numpy.array([])
-    f['foreground/fap_exc'] = numpy.array([])
+    f['foreground_h/h_iter' + h_iter_string + '/ifar'] = numpy.array([])
+    f['foreground_h/h_iter' + h_iter_string + '/fap'] = numpy.array([])
+    f['foreground_h/h_iter' + h_iter_string + '/ifar_exc'] = numpy.array([])
+    f['foreground_h/h_iter' + h_iter_string + '/fap_exc'] = numpy.array([])
+
+if 'name' in all_trigs.attrs:
+    f.attrs['name'] = all_trigs.attrs['name']
+
+# Incorporate hierarchical removal for any other loud triggers
+logging.info("Beginning hierarchical removal of foreground triggers.")
+
+# Step 1: Check to see if there are triggers that are louder
+#         then all of the inclusive background.
+
+# fnlouder is the number of background events that are louder than
+# the foreground trigger. So if a trigger is louder than all of
+# the background, fnlouder = 0.
+
+loud_t_ind = []
+for i in range(0, len(fnlouder)):
+    if fnlouder[i] == 0:
+        loud_t_ind.append(i)
+
+# Create arrays to save far and fap of hierarchically removed
+# foreground triggers
+
+save_ifar_fore_trig = numpy.array([], dtype=float)
```

Comment:
@ahnitz 
I've tried thinking about how to insert the FAR and FAP values back into the set of triggers in the loop. Using the method that I've sort of inherited from O1 removing triggers and associated timeslides, this seems pretty hard to do in the loop. I have to search iteratively backwards at how previous foreground triggers get removed in the list and decide where to properly put them back in the original list. Maybe I'm missing something obvious, but it seems easier to just do all the re-sorting at the end.

As far as I can tell the implementation in lines 425 to 494 will consistently place foreground triggers back where they sat in the original list of foreground triggers. If you really insist on that methodology I can think about how to do what you wanted.